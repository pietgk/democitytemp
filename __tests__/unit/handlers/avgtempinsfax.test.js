const lambda = require('../../../src/handlers/avgtempinsfax');
const { insertMockEvent } = require('../../../src/storage/storage-insert-mock-event');
const stage = 'test';
const httpMethod = 'GET';

describe('avgtempinsfax', () => {
    it('should fetch the avgtempinsfax from the database events collection', async () => {
        const _id = 'avgtempinsfax';
        const mockEvent = { _id, city: 'sfax', temperature: '12', unit: 'C' };
        const mockInsertResult = await insertMockEvent({ stage, ...mockEvent });

        const event = {
            httpMethod,
            stage
        };
        const responce = await lambda.getEventHandler(event);

        expect(mockInsertResult.insertedCount).toEqual(1);
        expect(mockInsertResult.insertedId).toEqual(_id);
        expect(responce.statusCode).toEqual(200);
        expect(responce.result).not.toEqual("null");
        expect(responce).toMatchSnapshot();
    });
});
