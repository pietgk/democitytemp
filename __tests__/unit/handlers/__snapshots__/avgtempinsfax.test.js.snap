// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avgtempinsfax should fetch the avgtempinsfax from the database events collection 1`] = `
Object {
  "body": "{\\"_id\\":\\"avgtempinsfax\\",\\"city\\":\\"sfax\\",\\"temperature\\":\\"12\\",\\"unit\\":\\"C\\"}",
  "statusCode": 200,
}
`;
